### ISSUE TITLE
Immediately Fires Discconnect Event, No Raw Output

### ISSUE THREAD

**Issue by BrandonCookeDev** [hydrabolt/discord.js#564]
I am trying to host my Discord bot on a Digital Ocean Linux server. 

This code works on Windows just fine, and there is not anything seemingly that would break the code in the transition. But nothing happens after login attempt other than the "Disconnected" event's code. No success function, no raw output. 

The code can be found in my repository 
https://github.com/BrandonCookeDev/DiscordCookieBot/blob/master/app.js in case you're interested,
but this seems like a bug. There's no proxy configuration I had to add for NPM so I don't believe it's a proxy issue.

**Comment by JMTK** [comment]
What's the nodejs version and Discord.js version on both?

**Comment by BrandonCookeDev** [comment]
I just upgraded it to the latest (on the server) Discord.js available from npm (8.1.0).
Node JS Version 6.3.0

**Comment by waterfoul** [comment]
I'm having a similar issue trying to deploy to a docker container based on node:6.4. You can deploy it using orderofperfectlight/discord-gw2-sync

**Comment by GusCaplan** [comment]
@waterfoul this isn't a fix per say, but if you aren't using autoReconnect it may at least reconnect the bot after whatever issue is causing it to DC. also, you can try adding `.catch(console.log)` to various parts of the bot, as well as using the `disconnect` event, which passes along errors that happen inside the client.

**Comment by waterfoul** [comment]
I'm already trying to catch the error event, it just never fires. Here is the pertinent code:
```
discord.bot.on('error', function(message) {
	console.log('error recieved', message);
});
```
I'll add a connect handler and see what that does to see if it's already auto-reconnecting (I didn't change the default)

**Comment by waterfoul** [comment]
Update: I've retooled my bot a bit to maintain 3 connections so it can work around this disconnect issue. I've got handlers logging warnings, errors, and disconnects. I appear to be getting an average of 4 "WS close: 1006" errors per client an hour so there is definetly something odd going on

**Comment by BrandonCookeDev** [comment]
Upgraded node to 6.5.0 and issue persists. I have yet to try downgrading.

**Comment by BrandonCookeDev** [comment]
Update: I switched to 4.5.0 and issue still persists. I'm really not sure what to do from here

**Comment by waterfoul** [comment]
I've found that my bot is still unstable even with the redundant connections

**Comment by hydrabolt** [comment]
Should be fixed in [9.0.2](https://github.com/hydrabolt/discord.js/releases/tag/9.0.2)

**Comment by BrandonCookeDev** [comment]
Would you mind explaining why this occurred? I would very much like to know.
